php/libraries/NDB_Page.class.inc

```diff
@@ -590,6 +595,83 @@ class NDB_Page
         return $this->form->createElement('password', $field, $label, $attr);
     }
 
+    /**
+     * An NDB_Page can act as both a middleware, or a handler, depending on the
+     * context. When treated as a middleware, the page is responsible for setting
+     * up any middleware required to access the page. By default, this will
+     * implement the page decoration middlewares which is responsible for adding
+     * the LORIS menu's/headers/footers/etc.
+     * This can be overridden by modules who don't want this behaviour (such
+     * as API endpoints.)
+     *
+     * @param ServerRequestInterface  $request The PSR7 request being processed.
+     * @param RequestHandlerInterface $handler The handler to handle the request
+     *                                         after processing the middleware.
+     *
+     * @return ResponseInterface the PSR15 response that was generated by the
+     * middleware
+     */
+    public function process(
+        ServerRequestInterface $request,
+        RequestHandlerInterface $handler
+    ) : ResponseInterface {
+        $baseURL = $request->getAttribute("baseurl");
+        $config  = NDB_Config::singleton();
+
+        $user = $request->getAttribute("user");
+
+        // Add the page class to the request to that middleware that needs it
```

Comment:
Not sure of what it mean

Reply:
The NDB_Page changes are inherited from #3477, can you make this comment over there?

Reply:
I added more context in #3477 (and rebased), is it clearer now?